src/components/Login/LoginMobileNumberCard.test.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  render,
+  fireEvent,
+  cleanup,
+  waitFor,
+} from "@testing-library/react-native";
+import React from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { AlertModalContextProvider } from "../../context/alert";
+import { CreateProvidersWrapper } from "../../test/helpers/providers";
+import { LoginMobileNumberCard } from "./LoginMobileNumberCard";
+import "../../common/i18n/i18nMock";
+
+jest.mock("../../utils/errorTracking");
+const mockCaptureException = jest.fn();
+(Sentry.captureException as jest.Mock).mockImplementation(mockCaptureException);
+
+const setLoginState = jest.fn();
+const setMobileNumber = jest.fn();
+const setCountryCode = jest.fn();
+const mockHandleRequestOTP = jest.fn().mockResolvedValue(true);
+
+describe("LoginMobileNumberCard", () => {
+  afterEach(() => {
+    cleanup();
+    jest.resetAllMocks();
+  });
+
+  it("should be able to fetch the endpoint without error", async () => {
+    expect.assertions(6);
+    const { getByTestId, queryByText } = render(
+      <LoginMobileNumberCard
+        setLoginStage={setLoginState}
+        setMobileNumber={setMobileNumber}
+        setCountryCode={setCountryCode}
+        handleRequestOTP={mockHandleRequestOTP}
+      />
+    );
+
+    const phoneNumberInput = getByTestId("login-phone-number-input");
+    const submitButton = getByTestId("login-send-otp-button");
```

Comment:
can move the testId `login-phone-number-input` and  `login-send-otp-button` to const variable.. easy to maintain at one place